gnomad_qc/v5/data_ingestion/federated_validity_checks.py

```diff
@@ -22,22 +179,101 @@ def main(args):
+            ht = test_ht.union(sex_ht)
+
+        # Read in parameters from config file.
+        with hl.hadoop_open(config_path, "r") as f:
+            config = json.load(f)
+
+        # Validate config file against schema.
+        try:
+            validate(instance=config, schema=schema)
+            logger.info("JSON is valid.")
+        except ValidationError as e:
+            raise ValueError(f"JSON validation error: %s, {e.message}")
+
+        missingness_threshold = config["missingness_threshold"]
+        indexed_array_annotations = config["indexed_array_annotations"]
+        struct_annotations_for_missingness = config[
+            "struct_annotations_for_missingness"
+        ]
+        freq_meta_expr = eval(config["freq_meta_expr"])
+        freq_annotations_to_sum = config["freq_annotations_to_sum"]
+        freq_sort_order = config["freq_sort_order"]
+        nhomalt_metric = config["nhomalt_metric"]
+
+        # Check that all neccesaary fields are present in the Table.
+        missing_fields = {}
+
+        # Check that specified global annotations are present.
+        global_fields = [i for i in indexed_array_annotations.values()]
+        missing_global_fields = [i for i in global_fields if i not in ht.globals]
+        missing_fields["globals"] = missing_global_fields
+
+        # Check that specified row annotations are present.
+        row_fields = [
+            i for i in indexed_array_annotations.keys()
+        ] + struct_annotations_for_missingness
+        missing_row_fields = [i for i in row_fields if i not in ht.row]
+        missing_fields["rows"] = missing_row_fields
+
+        # Check that freq_annotations_to_sum values are present in the 'freq' struct.
+        freq_field_names = list(ht.freq.dtype.element_type)
+        freq_annotations = freq_annotations_to_sum + [nhomalt_metric]
+        missing_freq_fields = [i for i in freq_annotations if i not in freq_field_names]
+        missing_fields["missing_freq_fields"] = missing_freq_fields
 
-        expected_contigs = [
-            i
-            for i in hl.get_reference("GRCh38").contigs
-            if i in [f"chr{i}" for i in range(1, 23)] + ["chrX", "chrY"]
+        # Check that sort_order values are present as keys within freq_meta_expr.
+        freq_meta_keys = set(
+            key for item in hl.eval(freq_meta_expr) for key in item.keys()
+        )
+        missing_sort_order_keys = [
+            i for i in freq_sort_order if i not in freq_meta_keys
         ]
-        logger.info("Summarizing variants and checking contigs.")
-        summarize_variants(ht, expected_contigs=expected_contigs)
+        missing_fields["missing_sort_order_keys"] = missing_sort_order_keys
+
+        if any(missing_fields.values()):
+            error_message = "Validation failed. Missing fields:\n" + "\n".join(
+                f"{key}: {value}" for key, value in missing_fields.items() if value
+            )
+            raise ValueError(error_message)
```

Comment:
Maybe pull all this out into its own check fields function to clean up main?

Reply:
Resolving, my original comment didnt show up during my final pass for some reason